clients/src/main/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManager.java

```diff
@@ -188,12 +184,12 @@ private void onFailedResponse(final long currentTimeMs, final Throwable exceptio
         if (exception == Errors.GROUP_AUTHORIZATION_FAILED.exception()) {
             log.debug("FindCoordinator request failed due to authorization error {}", exception.getMessage());
             KafkaException groupAuthorizationException = GroupAuthorizationException.forGroupId(this.groupId);
-            backgroundEventHandler.add(new ErrorEvent(groupAuthorizationException));
+            fatalError = Optional.of(groupAuthorizationException);
             return;
         }
 
         log.warn("FindCoordinator request failed due to fatal exception", exception);
-        backgroundEventHandler.add(new ErrorEvent(exception));
```

Comment:
I expect that removing this here will introduce a gap we need to cover: how are we going to get these coordinator errors from the consumer.poll now? (if no commit in use)
I see 2 options:
1. we keep the coordinator mgr propagating its errors via the event queue, just because we do need to know about them in the app thread (from the consumer.poll), without a triggering API event + the coordinator mgr exposes a `fatalError` just to allow other managers in the background thread take internal actions (ex. commit mgr needs to cancel pending requests).
2. the coord mgr does not propagate via error event anymore, just exposes the coordinator found or fatal error. Other managers that require a coordinator know what to do about it: 
    - commit request mgr uses the coordinator, or cancels all its pending requests if there is a coord fatal error
    - HB mgr uses the coordinator or propagates an error via ErrorEvent if there is a coord fatal error (currently the HB Mgr just skips sending a HB if there is no coordinator, but it could propagate the error too ) https://github.com/apache/kafka/blob/d76238a18fb6a86b4b08bc04918ea1a0ee626517/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java#L164

These are the only 2 managers that require a known coordinator, so I would expect we cover it all by ensuring that they handle the coordinator fatal error properly. Thoughts? (These are first thoughts, let's shape them)

Reply:
> we keep the coordinator mgr propagating its errors via the event queue, just because we do need to know about them in the app thread (from the consumer.poll)

In my opinion, if we remove the event queue propagation method in this PR, it might lead to unexpected behavior in SharedConsumer. If this concern exists, should we consider keeping the event queue propagation method?

> the coord mgr does not propagate via error event anymore, just exposes the coordinator found or fatal error. Other managers that require a coordinator know what to do about it

I'm not entirely clear on this approach, so I'm still thinking it through.

Reply:
It seems that each manager handles fatal errors differently, so allowing each manager to handle fatal errors in its own way after receiving them would be a better approach. I will draft a version to test this though.

Reply:
ack. With the approach of letting each manager that requires a coordinator handle the error as it needs, the only thing is probably what you mentioned that we need to handle it in the ShareHBMgr too right? (similar to how the ConsumerHBMgr does, sending ErrorEvent to the app thread?).

Reply:
> the only thing is probably what you mentioned that we need to handle it in the ShareHBMgr too right? (similar to how the ConsumerHBMgr does, sending ErrorEvent to the app thread?).

I wasn’t write clear, which caused some misunderstanding. I was considering whether `AsyncConsumer` and `ShareConsumer` handle fatal errors differently in `poll` method. However, at this point, it seems that moving the error event propagation to `AbstractHeartbeatRequestManager` should not be an issue.